impl/graphsync.go

```diff
@@ -256,13 +259,42 @@ func New(parent context.Context, network gsnet.GraphSyncNetwork,
 		ptqopts = append(ptqopts, peertaskqueue.MaxOutstandingWorkPerPeer(int(gsConfig.maxInProgressIncomingRequestsPerPeer)))
 	}
 	peerTaskQueue := peertaskqueue.New(ptqopts...)
-	responseManager := responsemanager.New(ctx, linkSystem, responseAssembler, peerTaskQueue, requestQueuedHooks, incomingRequestHooks, outgoingBlockHooks, requestUpdatedHooks, completedResponseListeners, requestorCancelledListeners, blockSentListeners, networkErrorListeners, gsConfig.maxInProgressIncomingRequests, network.ConnectionManager(), gsConfig.maxLinksPerIncomingRequest)
+	responseQueue := taskqueue.NewTaskQueue(ctx)
+	responseWorkSignal := make(chan struct{}, 1)
```

Comment:
Work signal is all already handled inside the TaskQueue -- generally, we shouldn't see it any were. You can remove all references to work signal from both ResponseManager and QueryExecutor.